pkg/autoscaler/statforwarder/processor.go

```diff
@@ -32,13 +32,36 @@ type bucketProcessor struct {
 	bkt string
 	// holder is the HolderIdentity for a bucket from the Lease.
 	holder string
-	// conn is the WebSocket connection to the holder pod.
-	conn *websocket.ManagedConnection
+
+	// podAddressable indicates whether pod is accessible with IP address.
+	podAddressable bool
+	svcDNS         string
+	// podConn is the WebSocket connection to the holder pod with pod IP.
+	podConn *websocket.ManagedConnection
+	// svcConn is the WebSocket connection to the holder pod with bucket Service.
+	svcConn *websocket.ManagedConnection
+
 	// `accept` is the function to process a StatMessage which doesn't need
 	// to be forwarded.
 	accept statProcessor
 }
 
+func newForwardProcessor(logger *zap.SugaredLogger, bkt, holder, podDNS, svcDNS string) *bucketProcessor {
+	logger.Info("Connecting to Autoscaler bucket at %s and %s.", podDNS, svcDNS)
+	// Initial with `podAddressable` true and a connection via IP address and a connection with SVC URL.
+	// If we create the connection when the stat arrives, the first request after the creation always
+	// fails because it takes some time to establish the connection.
+	return &bucketProcessor{
+		logger:         logger,
+		bkt:            bkt,
+		holder:         holder,
+		podAddressable: true,
+		podConn:        websocket.NewDurableSendingConnection(podDNS, logger),
+		svcConn:        websocket.NewDurableSendingConnection(svcDNS, logger),
```

Comment:
Should we create svc connection only if pod one failed?
Since in case of mesh the connect will fail outright, so there's no need to have two of those.

Reply:
I left a comment in the code why I did this:

```
// If we create the connection when the stat arrives, the first request after the creation always
// fails because it takes some time to establish the connection.
```

I think I am improving a case where 1) mesh is on 2) there're traffic when upgrading cluster. It's a trade-off between improving that specific case and wasting a connection for a period time. I'm not sure the percentage of Autoscaler pods with mesh. If that's is low, I'd change it to open pod connection only at the beginning.

Reply:
Isn't the connection establishment process sync?

Reply:
No. It's async in ManagedConnection.

Reply:
ok, changed to create pod connection only when initializing and use a short retry interval for the first 1 second.